### ISSUE TITLE
Any chance of adding CoinGecko?

### ISSUE THREAD

**Issue by jorijnsmit** [aneonex/BitcoinChecker#78]
Quite a nice API available here: https://www.coingecko.com/en/api.

**Comment by aneonex** [comment]
Why do you need CoinGecko?
Bitcoin Checker app interacts directly with exchanges API, but CoinGecko is not an exchange but a centralized aggregator.
Direct communication has various advantages: speed, availability, reliability, the ability to use the specifics of exchanges, and more.

**Comment by jorijnsmit** [comment]
Because it will add instant support for all coins listed there, including a lot of DEX. I saw you added Uniswap but there are uncountable other DEXes each with their own coins.

I'm not sure about alternatives, I know there is thegraph.com but it is also still missing a lot of exchanges.

Interesting problem, I'd be curious about other techniques/APIs of extracting price information from LPs for example! More and more of trading will take on-chain only.

**Comment by aneonex** [comment]
I agree, CoinGecko is very useful for extracting price information from DEX.  But when I added Uniswap, CoinGeko was missing information on some significant tokens, and there were other problems, so I had to use a more complex but reliable TheGraph.
Perhaps the situation has improved now.

It seems like a good idea to implement support for top 20-30 DEX through CoinGecko. Will this solve the issue?
Full integration with CoinGecko doesn't fit into the app's concept.

> Interesting problem, I'd be curious about other techniques/APIs of extracting price information from LPs for example! More and more of trading will take on-chain only.

Yes it's true. But what is "LPs"? :)

**Comment by jorijnsmit** [comment]
[Liquidity pools](https://youtu.be/cizLhxSKrAc). Problem is they don't use order books so price information is much more opaque. I find CoinGecko is somewhat of a solution here. This is their list of top DEXes: https://www.coingecko.com/en/dex.

**Comment by mbenatti** [comment]
:+1:  for this!
Will be awesome to have top 10/20/30 dex from coingecko on `BitcoinChecker`
In my point of view `LPs`  get out of the context of the APP, because the interface expect a pair(ex: `BTC/BUSD`)
For `LPs` we need to have a more explicit way to show this (as exemple, is often called `BTC/BUSD LP`)
An IDEA is to support the coins pair for these DEX on coingecko and put the `LP` on backlog for later

**Comment by aneonex** [comment]
@jorijnsmit Unfortunately, I was unable to find an API to get ticker for coin pair. CoinGecko only allows you to get a list of prices for a token, not for a pair. This leads to traffic overhead.
Instead of CoinGecko , I plan to use a TheGraph.com for all DEX, as already implemented for Uniswap.

**Comment by mbenatti** [comment]
> @jorijnsmit Unfortunately, I was unable to find an API to get ticker for coin pair. CoinGecko only allows you to get a list of prices for a token, not for a pair. This leads to traffic overhead.
> Instead of CoinGecko , I plan to use a TheGraph.com for all DEX, as already implemented for Uniswap.

any plans to have thegraph working?

**Comment by aneonex** [comment]
This turned out to be more difficult than I expected and will take longer, sorry. While the issue on hold.

**Comment by terrorist96** [comment]
@aneonex could you add pancakeswap similar to how you added uniswap in the meantime?
